src/Components/Footer.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { useCallback, useContext, useMemo } from 'react';
+import { TodosContext } from '../Context/TodosContext';
+import { TodosFilter } from './TodoFilter';
+
+export const Footer: React.FC = () => {
+  const { todos, setTodos } = useContext(TodosContext);
+
+  const todosLeft = useMemo(() => {
+    return todos.filter(todo => !todo.completed).length;
+  }, [todos]);
+
+  const removeCompletedTodos = useCallback(() => {
+    setTodos(todos.filter(prevTodo => !prevTodo.completed));
```

Comment:
```suggestion
    setTodos(todos.filter(todo => !todo.completed));
```
it's not previous todo actually, that's the current todo stored in todos array